presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -108,11 +115,22 @@ public synchronized boolean isWaitingForRevocableMemory()
     {
         checkArgument(bytes >= 0, "bytes is negative");
 
+        if (!memoryRevokingEnabled) {
+            checkLocalLimitExceeded(bytes);
+        }
+
         ListenableFuture<?> future = reserveMemoryUnsafe(bytes);
         revocableReserved += bytes;
         return future;
     }
 
+    private synchronized void checkLocalLimitExceeded(long bytes)
```

Comment:
put this method at the bottom of this file (near `willExceedsMemoryLimit`) and maybe rename them to something that will distinguish them?